crates/build/src/docker.rs

```diff
@@ -146,23 +122,43 @@ pub fn docker_build(args: ExecuteArgs) -> Result<BuildResult> {
             let _ = client.ping().await.map_err(|e| {
                 anyhow::anyhow!("{}\nIs your docker engine up and running?", e)
             })?;
-            let build_image =
-                get_image(client.clone(), image, &verbosity, &mut build_steps).await?;
 
-            run_build(
+            let image = match image {
+                ImageVariant::Custom(i) => i.clone(),
+                ImageVariant::Default => {
+                    format!("{}:{}", IMAGE, VERSION)
+                }
+            };
+
+            if let Err(err) =
+                pull_image(client.clone(), &image, &verbosity, &mut build_steps).await
+            {
+                // If the image could not be pulled, we will still attempt to use a local
+                // image of that name if it exists.
+                eprintln!(
+                    "{}",
+                    format!("Failed to pull the docker image {}: {}", image, err)
+                        .yellow()
+                        .bold(),
+                );
+            }
+
+            let build_result = run_build(
                 args,
-                &build_image,
-                &build_result_path,
+                &image,
                 &crate_metadata.contract_artifact_name,
                 &host_dir,
                 &verbosity,
                 &mut build_steps,
             )
             .await?;
 
-            let build_result = read_build_result(&host_dir, &build_result_path)?;
+            let metadata_artifacts = MetadataArtifacts {
+                dest_metadata: crate_metadata.metadata_path(),
```

Comment:
This will give only the relative path. Although there might be an CI config that rely on the absolute paths